File: 3075-maximize-happiness-of-selected-children/3075-maximize-happiness-of-selected-children.cpp
class Solution {
public:
    long long maximumHappinessSum(vector<int>& happiness, int k) {
        sort(happiness.begin(), happiness.end(), greater<int>());
        long long ans = 0;
      
        for(int i = 0; i<k; i++) {
            happiness[i] = max(0, happiness[i] - i);
            ans += happiness[i];
        }

        return ans;
    }
};

File: 3075-maximize-happiness-of-selected-children/3075-maximize-happiness-of-selected-children.py
class Solution(object):
    def maximumHappinessSum(self, happiness, k):
        happiness.sort(reverse = True)
        ans = 0

        for i in range(k):
            happiness[i] = max(0, happiness[i] - i)
            ans += happiness[i]
        
        return ans
        